saleor/graphql/attribute/utils.py

```diff
@@ -257,17 +257,33 @@ def _pre_save_reference_values(
             attribute.entity_type  # type: ignore
         ]
         get_or_create = attribute.values.get_or_create
-        return tuple(
-            get_or_create(
-                attribute=attribute,
-                slug=slugify(
-                    f"{instance.id}_{reference.id}",  # type: ignore
-                    allow_unicode=True,
-                ),
-                defaults={"name": getattr(reference, field_name)},
-            )[0]
-            for reference in attr_values.references
-        )
+
+        if attribute.entity_type == AttributeEntityType.PAGE:
+            return tuple(
+                get_or_create(
+                    attribute=attribute,
+                    reference_page=reference,
+                    slug=slugify(
+                        f"{instance.id}_{reference.id}",  # type: ignore
+                        allow_unicode=True,
+                    ),
+                    defaults={"name": getattr(reference, field_name)},
+                )[0]
+                for reference in attr_values.references
+            )
+        if attribute.entity_type == AttributeEntityType.PRODUCT:
+            return tuple(
+                get_or_create(
+                    attribute=attribute,
+                    reference_product=reference,
+                    slug=slugify(
+                        f"{instance.id}_{reference.id}",  # type: ignore
+                        allow_unicode=True,
+                    ),
+                    defaults={"name": getattr(reference, field_name)},
+                )[0]
+                for reference in attr_values.references
+            )
```

Comment:
We can simplify this. You can create dict with values  that are the same for `page` and `product` references and then update it with `reference_page` or `reference_product`. And at the end unpack it in the `get_or_create` method.